client/src/components/Search/Search.js

```diff
@@ -0,0 +1,64 @@
+import styled from "styled-components/macro";
+import PropTypes from "prop-types";
+import { useState } from "react";
+
+import { getRuns, searchForKm } from "../../api/runs";
+const Form = styled.form`
+  display: flex;
+  justify-content: center;
+
+  input {
+    height: 40px;
+    width: 70%;
+    margin-top: 15px;
+    border: solid 2px var(--secondary-color);
+    background: none;
+    color: var(--secondary-color);
+    font-family: "Heiti SC", adobe-heiti-std, sans-serif;
+    font-size: 1rem;
+    font-weight: 700;
+    display: flex;
+    text-align: center;
+    position: fixed;
+    top: 80px;
+  }
+`;
+
+function Search({ handleSubmit }) {
+  const [searchKm, setSearchKm] = useState();
+
+  return (
+    <Form
+      onSubmit={async (event) => {
+        event.preventDefault();
+        if (searchKm) {
+          handleSubmit(await searchForKm(searchKm));
+        } else {
+          handleSubmit(await getRuns());
+        }
+      }}
```

Comment:
I wouldn't expect to have the logic for this here, at least looking at the implementation of the component so far. However, after closer examination, this Form doesn't look like it would need to have its logic externally, because you want to handle the data and then do something with it, outside this component.
**TL;DR:** 😂 Anyway, consider moving this to a function for readability